docs/generate_cmd_doc.py

```diff
@@ -0,0 +1,182 @@
+    if p.returncode != 0:
+        logger.error("Error running {}.".format(cmd))
+        exit(1)
+    return p.stdout
+
+
+def get_description_from_output(text: str) -> str:
+    pattern = r"Usage:\n.*?\n\n(.*?)\n.*"
+    match = re.search(pattern, text, re.DOTALL)
+    if match:
+        desired_block = match.group(1).strip()
+        return desired_block
+    return ""
+
+
+def render_title(text: str, cmd: str) -> str:
+    text = re.sub(
+        r"\(reference_pebble_cmd_command\)=",
+        f"(reference_pebble_{cmd}_command)=",
+        text,
+        flags=re.MULTILINE,
+    )
+    return re.sub(
+        r"^# cmd command$",
+        f"# {cmd} command",
+        text,
+        flags=re.MULTILINE,
+    )
+
+
+def render_description(text: str, description: str) -> str:
+    return re.sub(r"^Description$", f"{description}", text, flags=re.MULTILINE)
+
+
+def render_code_block_cmd(text: str, cmd: str) -> str:
+    return re.sub(r"(:input: ).*$", rf"\1{cmd}", text, count=1, flags=re.MULTILINE)
+
+
+def render_code_block_output(text: str, output: str) -> str:
+    start_pos = text.find(AUTOMATED_START_MARKER)
+    end_pos = text.find(AUTOMATED_STOP_MARKER) + len(AUTOMATED_STOP_MARKER)
+    return text[:start_pos] + output + text[end_pos:]
+
+
+def update_toc(all_cmds: typing.List[str]):
+    index_page = "reference/cli-commands/cli-commands.md"
+    with open(index_page, "r") as file:
+        text = file.read()
+
+    start_index = text.find("```{toctree}")
+    end_index = text.find("```", start_index + 1) + 3
+    cmd_list = "\n".join(["{cmd} <{cmd}>".format(cmd=cmd) for cmd in all_cmds])
+    toc_tree = """\
+```{{toctree}}
+:titlesonly:
+:maxdepth: 1
+
+{cmd_list}
+```""".format(cmd_list=cmd_list)
+
+    text = text[:start_index] + toc_tree + text[end_index:]
+    with open(index_page, "w") as file:
+        file.write(text)
+
+
+def git_diff() -> int:
+    cmd = "git diff --exit-code"
+    p = subprocess.run(cmd.split())
+    if p.returncode != 0:
+        logger.warning("Docs are updated, please commit them.")
+    return p.returncode
+
+
+def create_file_if_not_exist(filepath: str, cmd: str) -> bool:
+    file_existed = os.path.exists(filepath)
+    if not file_existed:
+        logger.info(
+            "The doc for command {} doesn't exist, "
+            "creating from the template.".format(cmd)
+        )
+        with open(filepath, "w") as file:
+            file.write(TEMPLATE)
+    return file_existed
+
+
+def generate_help_command_and_output(cmd: str) -> typing.Tuple[str, str]:
+    help_cmd = "pebble {} --help".format(cmd)
+
+    help_cmd_output = """\
+<!-- START AUTOMATED OUTPUT -->
+```{{terminal}}
+:input: {help_cmd}
+{stdout}
+```
+<!-- END AUTOMATED OUTPUT -->""".format(
+        help_cmd=help_cmd,
+        stdout=get_command_help_output(help_cmd).strip(),
+    )
+
+    return help_cmd, help_cmd_output
+
+
+def process_command(cmd: str):
+    logger.info("Processing doc for command {}".format(cmd))
```

Comment:
It's standard [when logging to use `%`-formatting](https://docs.python.org/3/howto/logging.html#logging-variable-data), as it `logger.info()` and similar functions lazily-format the arguments. Or you can use f-strings, but `%` is probably most standard.

```
logger.info('Processing doc for command %s', cmd)
```